### PR TITLE
Powershell scripts to support installation from source.

### PR DESCRIPTION
**Pull request by paninaro**

### CODE REVIEW

**Review by hoffmang9 — DISMISSED** [review]
lgtm

File: Install-gui.ps1 [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    Write-Host "Unable to find Node.js"
+    Exit 1
+}
+
+Write-Host "Running git submodule update --init --recursive."
+Write-Host ""
+git submodule update --init --recursive
+Write-Host "Running git submodule update."
+Write-Host ""
+git submodule update
```
Comment (harold-b) [inline Install-gui.ps1:21]: Shouldn't the first `update` call be sufficient?

Comment (paninaro) [reply Install-gui.ps1:21]: I'm not sure why install-gui.sh has this line. Shouldn't be necessary. I'll remove it from this script.

**Comment by jack60612** [comment]
@paninaro did you mean to include the gui updates?

**Comment by paninaro** [comment]
> @paninaro did you mean to include the gui updates?

Ack, thanks for the catch. That slipped past me.

**Comment by jack60612** [comment]
@paninaro once you fix the update thing i'll be happy to approve

**Comment by jack60612** [comment]
> @paninaro once you fix the update thing i'll be happy to approve

never mind lol

**Review by jack60612 — DISMISSED** [review]
LGTM

**Comment by jack60612** [comment]
trying to get this squeezed in for 1.2.8

**Review by altendky — COMMENTED** [review]
I don't see any issues, though I have limited PowerShell experience.  All comments here and inline are just things I think might be nice for us or everyone else.

When I am adding suggestions it is weirdly adding an extra blank line into them.  Are these files using `\r\n` newlines?  Do we want that?  I don't know if PowerShell cares, but if not then I personally would stick with `\n`.  Also, apparently it got a BOM added and is thus UTF-8 instead of ASCII?  Doesn't super matter, just a bit odd.  `:]`

```
Install.ps1: UTF-8 Unicode (with BOM) text, with CRLF line terminators
```

A bit of a nitpick, but maybe keeping lowercase names would be good so they sort with the related Linux files?

image

File: Install-gui.ps1 [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿$ErrorActionPreference = "Stop"
+
+if ($null -eq (Get-ChildItem env:VIRTUAL_ENV -ErrorAction SilentlyContinue))
+{
+    Write-Output "This requires the chia python virtual environment."
```
[the remark this answers is not in the document]
Comment (paninaro) [reply Install-gui.ps1]: Updated with text similar to the suggestion

File: Install-gui.ps1 [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+$ErrorActionPreference = "SilentlyContinue"
+npm install --loglevel=error
+npm audit fix
+npm run build
+py ..\installhelper.py
+
+Write-Output ""
+Write-Output "Chia blockchain Install-gui.ps1 completed."
+Write-Output ""
+Write-Output "Type 'cd chia-blockchain-gui' and then 'npm run electron' to start the GUI."
```
[the remark this answers is not in the document]
Comment (paninaro) [reply Install-gui.ps1]: Done

File: Install-gui.ps1 [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    Exit 1
+}
+
+if ($null -eq (Get-Command node -ErrorAction SilentlyContinue))
+{
+    Write-Output "Unable to find Node.js"
+    Exit 1
+}
+
+Write-Output "Running git submodule update --init --recursive."
```
[the remark this answers is not in the document]
Comment (paninaro) [reply Install-gui.ps1]: Done

File: Install.ps1 [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+{
+    Write-Output "Unable to find git"
+    Exit 1
+}
+
+git submodule update --init mozilla-ca
+
+if ($null -eq (Get-Command py -ErrorAction SilentlyContinue))
+{
+    Write-Output "Unable to find py"
```
[the remark this answers is not in the document]
Comment (paninaro) [reply Install.ps1]: Done

**Comment by paninaro** [comment]
> I don't see any issues, though I have limited PowerShell experience. All comments here and inline are just things I think might be nice for us or everyone else.
> 
> When I am adding suggestions it is weirdly adding an extra blank line into them. Are these files using `\r\n` newlines? Do we want that? I don't know if PowerShell cares, but if not then I personally would stick with `\n`. Also, apparently it got a BOM added and is thus UTF-8 instead of ASCII? Doesn't super matter, just a bit odd. `:]`
> 
> ```
> Install.ps1: UTF-8 Unicode (with BOM) text, with CRLF line terminators
> ```
> 
> A bit of a nitpick, but maybe keeping lowercase names would be good so they sort with the related Linux files?
> 
> image

Updated the encoding to UTF-8 without a BOM. Also changed the line-endings to LF. Git should handle the line ending conversion automatically on checkout.

I kept the naming capitalized as that appears to be a PowerShell convention/best practice.

File: Install.ps1 [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    Write-Output "Note the check box during installation of Python to install the Python Launcher for Windows."
+    Write-Output ""
+    Write-Output "https://docs.python.org/3/using/windows.html#installation-steps"
+    Exit 1
+}
+
+$pythonVersion = (py --version).split(" ")[1]
+if ([version]$pythonVersion -lt [version]"3.7.0")
+{
+    Write-Output "Found Python version:" $pythonVersion
```
Comment (altendky) [inline Install.ps1:38]: I was thinking immediately above the `if` so it was shown in both cases.  Not a big deal though.

**Review by altendky — APPROVED** [review]

**Review by jack60612 — APPROVED** [review]
lgtm

**Review by harold-b — APPROVED** [review]

**Pull request merged**